yt/visualization/base_plot_types.py

```diff
@@ -341,7 +341,20 @@ def _set_axes(self) -> None:
         self.image.axes.set_facecolor(self.colorbar_handler.background_color)
 
         self.cax.tick_params(which="both", direction="in")
-        self.cb = self.figure.colorbar(self.image, self.cax)
+
+        # For creating a multipanel plot by ImageGrid
+        # we may need the location keyword, which was introduced since Matplotlib 3.7.0
+        cb_location = getattr(self.cax, "orientation", None)
+        if matplotlib.__version_info__ >= (3, 7):
+            cb_kwargs = {"location": cb_location}
+        else:
+            cb_kwargs = {}
+            if cb_location in ["top", "bottom"]:
+                warnings.warn(
+                    "Colorbar orientation would be wrong in the current Matplotlib version (< 3.7.0)",
+                    stacklevel=2,
```

Comment:
since we are within a private function, I think the correct stacklevel should be at least 3, but we should check that in the case this warning is triggered, it really points to the last line of user code (a manual check is enough for this).
Independently, we should also have a test for this warning, to see that it pops up in the expected conditions.

Reply:
> but we should check that in the case this warning is triggered, it really points to the last line of user code (a manual check is enough for this).

In this case, I think it should be set to 6, like this.

```
/var/folders/kw/_83g0dzx6v968ryzq81mjm3h0000gn/T/ipykernel_9325/1071457781.py:12: UserWarning: Colorbar orientation would be wrong in the current Matplotlib version (< 3.7.0)
  fig = p.export_to_mpl_figure((2, 2), cbar_location="top", cbar_mode="edge", axes_pad=0)

```

> Independently, we should also have a test for this warning, to see that it pops up in the expected conditions.

So which file do you think would be more appropriate to add this test to? `test_image_compare_2D_plots.py` seems to be used only for image tests.

Reply:
I would have recommended putting the test next to other tests exercising `export_to_mpl_figure` ... but then discovered there are currently none ! so, even though this is indeed slightly off-topic for `test_image_compare_2D_plots.py`, it's still the most natural place to put the new test I think.